geospaas_harvesting/ingesters.py

```diff
@@ -139,8 +145,8 @@ def _ingest_dataset(self, url, normalized_attributes):
                 if params.count() > 1 and units is not None:
                     params = params.filter(units=units)
                 if params.count() >= 1:
-                    dsp, dsp_created = DatasetParameter.objects.get_or_create(
-                            dataset=dataset, parameter=params[0])
+                    DatasetParameter.objects.get_or_create(
```

Comment:
while you are at it, can you replace the multiple `params.count()` calls by a variable assignment and then use this variable?

Reply:
Does it improve the performance of code or you suggest it only base on writing style?

Reply:
For performance: it is less costly to call a function once rather than multiple times.

Reply:
I guess that after reading your suggestion! ok I will do it.

Reply:
As we discussed, the multiple calls are necessary here, my bad

Reply:
Discussed and Resolved!!!! without changing!